### PR TITLE
Run instance groups on deployment manifest creation and fix tests

### PR DESCRIPTION
**Pull request by viovanov**
[#164072732](https://www.pivotaltracker.com/story/show/164072732)

### CODE REVIEW

File: pkg/kube/controllers/extendedjob/errand_reconciler.go [hunk 1]
```diff
@@ -107,6 +107,11 @@ func (r *ErrandReconciler) Reconcile(request reconcile.Request) (result reconcil
 }
 
 func (r *ErrandReconciler) createJob(ctx context.Context, extJob ejv1.ExtendedJob) error {
+	if extJob.Spec.Template.Labels == nil {
+		extJob.Spec.Template.Labels = map[string]string{}
+	}
+	extJob.Spec.Template.Labels["ejob-name"] = extJob.Name
```
Comment (manno) [inline pkg/kube/controllers/extendedjob/errand_reconciler.go]: I don't understand this code. It modifies the extended job and relies on later code to store this?
Are labels transferred from the extended job to the pod somehow, or how does clean up code find pods with that label?

Could this be moved to be near the actual `Update(extjob)` statement, it's not  obvious from naming and function signature, that the passed in `extJob` is mutated here.

File: pkg/kube/controllers/extendedjob/trigger_reconciler.go [hunk 2]
```diff
@@ -127,6 +127,11 @@ func (r *TriggerReconciler) Reconcile(request reconcile.Request) (result reconci
 }
 
 func (r *TriggerReconciler) createJob(ctx context.Context, extJob ejv1.ExtendedJob, podName string) error {
+	if extJob.Spec.Template.Labels == nil {
+		extJob.Spec.Template.Labels = map[string]string{}
+	}
+	extJob.Spec.Template.Labels["ejob-name"] = extJob.Name
```
Comment (manno) [inline pkg/kube/controllers/extendedjob/trigger_reconciler.go]: Same as errand reconciler comment

File: integration/environment/environment.go [hunk 3]
```diff
@@ -52,7 +55,7 @@ func NewEnvironment() *Environment {
 			Fs:         afero.NewOsFs(),
 		},
 		Machine: Machine{
-			pollTimeout:  30 * time.Second,
+			pollTimeout:  60 * time.Second,
```
Comment (manno) [inline integration/environment/environment.go]: Is it necessary to increase the timeout?

File: integration/extended_job_test.go [hunk 4]
```diff
@@ -194,33 +195,34 @@ var _ = Describe("ExtendedJob", func() {
...
 			tearDown, err = env.CreatePod(env.Namespace, pod)
 			Expect(err).NotTo(HaveOccurred())
-			defer tearDown()
+			defer func(tdf environment.TearDownFunc) { Expect(tdf()).To(Succeed()) }(tearDown)
 
 			By("waiting for the job")
 			_, err = env.CollectJobs(env.Namespace, "extendedjob=true", 1)
 			Expect(err).NotTo(HaveOccurred(), "error waiting for jobs from extendedjob")
 
 			// Why does the ExtendedJob deletion not trigger the job deletion?
```
Comment (manno) [inline integration/extended_job_test.go]: This comment can probably be removed now?

File: testing/catalog.go [hunk 5]
```diff
@@ -572,9 +692,8 @@ func (c *Catalog) AnnotatedPod(name string, annotations map[string]string) corev
 
 // Sleep1hPodSpec defines a simple pod that sleeps 60*60s for testing
 func (c *Catalog) Sleep1hPodSpec() corev1.PodSpec {
-	one := int64(1)
 	return corev1.PodSpec{
-		TerminationGracePeriodSeconds: &one,
+		TerminationGracePeriodSeconds: helper.Int64(1),
```
Comment (manno) [inline testing/catalog.go]: We should replace all occurrences with the helper, not just this one.

File: integration/deploy_test.go [hunk 6]
```diff
@@ -3,26 +3,28 @@ package integration_test
...
 		AfterEach(func() {
-			env.WaitForPodsDelete(env.Namespace)
+			Expect(env.WaitForPodsDelete(env.Namespace)).To(Succeed())
 		})
 
 		It("should deploy a pod with 10 seconds for the reconciler context", func() {
 			tearDown, err := env.CreateConfigMap(env.Namespace, env.DefaultBOSHManifestConfigMap("manifest"))
 			Expect(err).NotTo(HaveOccurred())
-			defer tearDown()
+			defer func(tdf environment.TearDownFunc) { Expect(tdf()).To(Succeed()) }(tearDown)
```
Comment (manno) [inline integration/deploy_test.go]:
```suggestion
			defer func(f environment.TearDownFunc) { Expect(f()).To(Succeed()) }(tearDown)
```
I think just `f` would be just fine here.

File: Dockerfile-nobuild [hunk 7]
```diff
@@ -0,0 +1,4 @@
+FROM opensuse/leap:15.0
+RUN zypper -n in system-user-nobody
+USER nobody
+COPY binaries/cf-operator /usr/local/bin/cf-operator
```
Comment (manno) [inline Dockerfile-nobuild]: I guess it's not possible to do this in the same dockerfile.  `RUN if` maybe?

File: pkg/kube/controllers/extendedjob/errand_reconciler.go (hunk 1 above)
Comment (viovanov) [reply pkg/kube/controllers/extendedjob/errand_reconciler.go]: About moving - I agree about not wanting to mutate it here.
About the other comments, I don't know what you mean. The label will exist on the pod. That's it - it's required to link it back to the ejob. `job-name` is insufficient.

File: integration/environment/environment.go (hunk 3 above)
Comment (viovanov) [reply integration/environment/environment.go]: nope, forgot to switch it back 👍

File: integration/extended_job_test.go (hunk 4 above)
Comment (viovanov) [reply integration/extended_job_test.go]: 👍

File: testing/catalog.go (hunk 5 above)
Comment (viovanov) [reply testing/catalog.go]: 👍

File: Dockerfile-nobuild (hunk 7 above)
Comment (viovanov) [reply Dockerfile-nobuild]: I don't think what you mean is possible.

File: pkg/kube/controllers/extendedjob/errand_reconciler.go (hunk 1 above)
Comment (manno) [reply pkg/kube/controllers/extendedjob/errand_reconciler.go]: Yes I missed that the label is set on `Template` which is the actual `PodSpec` being for the pod.

Comment (viovanov) [reply pkg/kube/controllers/extendedjob/errand_reconciler.go]: 👍

File: testing/catalog.go (hunk 5 above)
Comment (viovanov) [reply testing/catalog.go]: https://www.pivotaltracker.com/story/show/164729001

File: pkg/kube/controllers/extendedjob/errand_reconciler.go (hunk 1 above)
Comment (manno) [reply pkg/kube/controllers/extendedjob/errand_reconciler.go]: Ah, I see now, we don't actually need to mutate extendjob (not even sure if it does). We just need to pass this modified PodTemplateSpec to Job a few lines below

File: bin/up [hunk 8]
```diff
@@ -9,6 +9,12 @@ if type -a minikube >/dev/null 2>/dev/null; then
     eval ${dockerenv}
 fi
 
+GIT_ROOT="${GIT_ROOT:-$(git rev-parse --show-toplevel)}"
```
Comment (manno) [inline bin/up]: I just noticed this introduces `GIT_ROOT` to have the absolute path available, but most of the script still uses relative paths (line 20, ...).
We could remove `GIT_ROOT` here and in `bin/test-integration`, too.

**Pull request merged**